src/IdentityServer/Licensing/LicenseValidator.cs

```diff
@@ -15,41 +15,48 @@
 namespace Duende;
 
 // shared APIs needed for Duende license validation
-internal partial class LicenseValidator
+internal class LicenseValidator<T>
+    where T : License, new()
 {
     static readonly string[] LicenseFileNames = new[] 
     {
         "Duende_License.key",
         "Duende_IdentityServer_License.key",
     };
 
-    static ILogger _logger;
-    static Action<string, object[]> _errorLog;
-    static Action<string, object[]> _informationLog;
-    static Action<string, object[]> _warningLog;
-    static Action<string, object[]> _debugLog;
+    protected ILogger Logger;
+    protected Action<string, object[]> ErrorLog;
+    protected Action<string, object[]> InformationLog;
+    protected Action<string, object[]> WarningLog;
+    protected Action<string, object[]> DebugLog;
 
-    static License _license;
+    protected T License { get; private set; }
+    
+    // cloned copy meant to be accessible in DI
+    T _copy;
+    public T GetLicense()
+    {
+        if (_copy == null && License != null)
+        {
+            _copy = new T();
+            _copy.Initalize(License.Claims.Clone());
+        }
+        return _copy;
+    }
 
-    static void Initalize(ILoggerFactory loggerFactory, string productName, string key, bool isDevelopment = false)
+    protected void Initalize(ILoggerFactory loggerFactory, string productName, string key)
     {
-        _logger = loggerFactory.CreateLogger($"Duende.{productName}.License");
+        //if (Logger != null) throw new InvalidOperationException("LicenseValidator already initialized.");
```

Comment:
Do we want this commented out?

Reply:
I removed it because I don't think it hurts to re-run it and I thought it might be possible that people might call this more than once for some reason.